src/fileio.js

```diff
@@ -0,0 +1,113 @@
+const log4js = require('log4js');
+const fs = require('fs');
+const csvToJs = require('csv-parse/lib/sync');
+const xmlToJs = require('xml-js').xml2js;
+const moment = require('moment');
+
+const Account = require('./data.js');
+
+const logger = log4js.getLogger('parser.js');
+
+function parseFile(fileName, accounts) {
+    let file;
+    try {
+        file = fs.readFileSync(fileName, {encoding: 'utf8'});
+    } catch (error) {
+        console.log(`${fileName} couldn't be read: ${error.message}`)
+        logger.warn(error);
+        return;
+    }
+    if (!file) {
+        logger.fatal(`${fileName} couldn't be read, obtained ${file}`);
+        console.log(`The supplied file ${fileName} couldn't be read.`);
+        process.exit(1);
+    }
+
+    const parsers = {
+        'csv': parseCSV,
+        'json': parseJSON,
+        'xml': parseXML,
+    }
+    const extension = fileName.substring(fileName.lastIndexOf('.') + 1);
+    if (!parsers[extension]) {
+        logger.warn(`Failed to import file ${fileName} unsupported type.`);
+        console.log('Unsupported filetype.');
+        return;
+    }
+    logger.info(`Importing file ${fileName} detected type ${extension}.`);
+    const records = parsers[extension](file);
+
+    for (let i = 0; i < records.length; i++) {

```

Comment:
No reason for this to not be `records.forEach` (I think? Doesn't look like you use `i` for anything other than getting the record.)